test/functional/eval/timer_spec.lua

```diff
@@ -19,72 +19,85 @@ describe('timers', function()
   it('are triggered during sleep', function()
-    command("call timer_start(50, 'MyHandler', {'repeat': 2})")
+    source([[
+      let g:val = -1
+      func! MyHandler(timer)
+        if g:val >= 0
+          let g:val += 1
+          if g:val == 2
+            call timer_stop(a:timer)
+          endif
+        endif
+      endfunc
+    ]])
+    eval("timer_start(10, 'MyHandler', {'repeat': -1})")
     nvim_async("command", "sleep 10")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(300))
-    eq(2,eval("g:val"))
+    eq(-1, eval("g:val"))  -- timer did nothing yet.
+    nvim_async("command", "let g:val = 0")
+    run(nil, nil, nil, load_adjust(20))
+    retry(nil, nil, function()
+      eq(2, eval("g:val"))
+    end)
   end)
 
   it('works with zero timeout', function()
     -- timer_start does still not invoke the callback immediately
-    eq(0,eval("[timer_start(0, 'MyHandler', {'repeat': 1000}), g:val][1]"))
-    run(nil, nil, nil, load_adjust(400))
-    eq(1000,eval("g:val"))
+    eq(0, eval("[timer_start(0, 'MyHandler', {'repeat': 1000}), g:val][1]"))
+    retry(nil, nil, function()
+      eq(1000, eval("g:val"))
+    end)
   end)
 
   it('can be started during sleep', function()
     nvim_async("command", "sleep 10")
     -- this also tests that remote requests works during sleep
-    eval("timer_start(50, 'MyHandler', {'repeat': 2})")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(300))
-    eq(2,eval("g:val"))
+    eq(0, eval("[timer_start(10, 'MyHandler', {'repeat': 2}), g:val][1]"))
+    run(nil, nil, nil, load_adjust(20))
+    retry(nil, load_adjust(300), function() eq(2,eval("g:val")) end)
   end)
 
   it('are paused when event processing is disabled', function()
-    command("call timer_start(50, 'MyHandler', {'repeat': -1})")
-    run(nil, nil, nil, load_adjust(100))
+    command("call timer_start(5, 'MyHandler', {'repeat': -1})")
+    run(nil, nil, nil, load_adjust(10))
     local count = eval("g:val")
     -- shows two line error message and thus invokes the return prompt.
     -- if we start to allow event processing here, we need to change this test.
     feed(':throw "fatal error"<CR>')
-    run(nil, nil, nil, load_adjust(300))
+    run(nil, nil, nil, load_adjust(30))
     feed("<cr>")
     local diff = eval("g:val") - count
     assert(0 <= diff and diff <= 4,
            'expected (0 <= diff <= 4), got: '..tostring(diff))
   end)
 
   it('are triggered in blocking getchar() call', function()
-    command("call timer_start(50, 'MyHandler', {'repeat': -1})")
-    nvim_async("command", "let g:c = getchar()")
-    run(nil, nil, nil, load_adjust(300))
+    command("call timer_start(5, 'MyHandler', {'repeat': -1})")
+    nvim_async("command", "let g:val = 0 | let g:c = getchar()")
+    retry(nil, nil, function()
+      ok(eval("g:val") >= 2)
+      eq(0, eval("getchar(1)"))
+    end)
     feed("c")
-    local count = eval("g:val")
-    assert(count >= 3, 'expected count >= 3, got: '..tostring(count))
```

Comment:
the failure messages for `ok()` are useless, that's why assert() is used like that.

Reply:
ok (pun intended) - will fix this later.
Should we discourage of "ok" then?

Reply:
> Should we discourage of "ok" then?

Not sure. Maybe it could take an optional message parameter.

Reply:
=> https://github.com/neovim/neovim/pull/10820